storage_test.go

```diff
@@ -0,0 +1,442 @@
+			len(fixturesWithPrefix), len(foundKeys))
+		return
+	}
+
+	noKeysFound, err := storage.List("invalid", false)
+	if err != nil {
+		t.Errorf("unable to list keys with invalid prefix: %s", err.Error())
+		return
+	}
+
+	if len(noKeysFound) != 0 {
+		t.Errorf("should not have found any keys but found %v key", len(noKeysFound))
+		return
+	}
+}
+
+func TestDynamoDBStorage_Stat(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	storage := Storage{
+		Table:         TestTableName,
+		AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+		AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+		AwsDisableSSL: DisableSSL,
+	}
+
+	err = storage.Store("key", []byte("value"))
+	if err != nil {
+		t.Errorf("failed to store fixture key/value: %s", err.Error())
+		return
+	}
+
+	stat, err := storage.Stat("key")
+	if err != nil {
+		t.Errorf("failed to stat item: %s", err.Error())
+		return
+	}
+
+	if stat.Key != "key" {
+		t.Errorf("stat key does not match expected. got: %s", stat.Key)
+		return
+	}
+	if stat.Size != int64(len("value")) {
+		t.Errorf("stat size does not match expected. got: %v", stat.Size)
+		return
+	}
+	if time.Since(stat.Modified) > 5*time.Second {
+		t.Errorf("stat modified time is not within 3 seoncds. got: %s", stat.Modified)
+		return
+	}
+}
+
+func TestDynamoDBStorage_Delete(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	storage := Storage{
+		Table:         TestTableName,
+		AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+		AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+		AwsDisableSSL: DisableSSL,
+	}
+
+	err = storage.Store("key", []byte("value"))
+	if err != nil {
+		t.Errorf("failed to store fixture key/value: %s", err.Error())
+		return
+	}
+
+	value, err := storage.Load("key")
+	if err != nil {
+		t.Errorf("unable to load key that was just stored: %s", err.Error())
+		return
+	}
+
+	if string(value) != "value" {
+		t.Errorf("value returned does not match expected. expected: %s, got: %s", "value", string(value))
+		return
+	}
+
+	err = storage.Delete("key")
+	if err != nil {
+		t.Errorf("unable to delete key: %s", err.Error())
+		return
+	}
+
+	if storage.Exists("key") {
+		t.Errorf("key still exists after delete")
+		return
+	}
+
+}
+
+func TestDynamoDBStorage_Lock(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	storage := Storage{
+		Table:         TestTableName,
+		AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+		AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+		AwsDisableSSL: DisableSSL,
+		LockTimeout:   6 * time.Second,
+	}
+
+	// create lock
+	key := "test1"
+	err = storage.Lock(key)
+	if err != nil {
+		t.Errorf("error creating lock: %s", err.Error())
+	}
+
+	// try to create lock again, it should take about 5-10 seconds to return
+	before := time.Now()
+	err = storage.Lock(key)
+	if err != nil {
+		t.Errorf("error creating lock second time: %s", err.Error())
+	}
+	if time.Since(before) < 5*time.Second {
```

Comment:
This should match line 396, right? Would it be worth making it a shorter time, in order to avoid having a test that takes so long?

Reply:
sure, I meant to shorten it up and forgot.